### PR TITLE
MoreVP flow

### PR DESCRIPTION
**Pull request by paulperegud**
Please review, but do not merge yet.

Done:
* all the things, branch `dev-morevp` (by Kelvin)
* fixing discrepancies between `master` and `dev-morevp`
* ERC20 tokens in transaction format
* bug which caused mature in-flight exits to be processed after not-yet-mature standard exits

TODO:
* ERC20 handling for in-flight exits
* ~~high exit priority for deposits (the one and only failing test ATM)~~ DONE
* ~~fee exits~~ DONE

### CODE REVIEW

**Comment by kfichter** [comment]
> bug which caused mature in-flight exits to be processed after not-yet-mature standard exits

What caused this? Just curious.

**Comment by paulperegud** [comment]
Most significant bit of priority uint256 value set to 1 for in-flight exits. This changed the order from one sorted by {exitable_timestamp} to one sorted by {type, exitable_timestamp}. To fix this, bits were changed from:

* use bits [0-191] as exit id (192 bits)
* use bits [192-254] as exitable timestamp (63 bits)
* use most-significant bit [255] as a in-flight flag (1 bit)

to:
* use least-significant bit [0] as a in-flight flag (1 bit)
* use bits [1-191] as exit id (191 bits)
* use bits [255-192] as exitable timestamp (64 bits)

File: testlang/testlang.py [hunk 1]
```diff
@@ -160,12 +175,14 @@ def spend_utxo(self, input_ids, keys, outputs=[], force_invalid=False):
         spend_id = encode_utxo_id(blknum, 0, 0)
         return spend_id
 
-    def start_standard_exit(self, output_id, key, bond=None):
+    def start_standard_exit(self, output_id, key, bond=None, sender=None):
```
[the remark this answers is not in the document]
Comment (paulperegud) [reply testlang/testlang.py]: Done

File: tests/contracts/rlp/test_plasma_core.py [hunk 2]
```diff
@@ -24,9 +24,9 @@ def test_slice_signature(plasma_core_test):
 def test_get_output(plasma_core_test):
     owner = b'0x82a978b3f5962a5b0957d9ee9eef472ee55b42f1'
     amount = 100
-    tx = Transaction(outputs=[(owner, amount)])
-    assert plasma_core_test.getOutput(tx.encoded, 0) == [owner.decode("utf-8"), amount]
-    assert plasma_core_test.getOutput(tx.encoded, 1) == ['0x0000000000000000000000000000000000000000', 0]
+    tx = Transaction(outputs=[(owner, '0x0000000000000000000000000000000000000000', amount)])
```
[the remark this answers is not in the document]
Comment (paulperegud) [reply tests/contracts/rlp/test_plasma_core.py]: Done

File: tests/contracts/root_chain/test_start_standard_exit.py [hunk 3]
```diff
@@ -97,22 +96,30 @@ def test_start_standard_exit_wrong_oindex_should_fail(testlang):
...
 
     with pytest.raises(TransactionFailed):
-        testlang.start_standard_exit(alice, bob_utxo)
+        testlang.start_standard_exit(bob_utxo, alice.key)
 
     with pytest.raises(TransactionFailed):
-        testlang.start_standard_exit(bob, alice_utxo)
+        testlang.start_standard_exit(alice_utxo, bob.key)
+
+    testlang.start_standard_exit(alice_utxo, alice.key)
```
[the remark this answers is not in the document]
Comment (paulperegud) [reply tests/contracts/root_chain/test_start_standard_exit.py]: Those two lines not throwing are a good indication that the rest of the test is correctly constructed.

File: plasma_core/block.py [hunk 4]
```diff
@@ -43,10 +43,12 @@ def encoded(self):
...
 
     def sign(self, key):
-        self.sig = sign(self.hash, key)
+        self.signature = sign(self.hash, key)
 
 
-UnsignedBlock = Block.exclude(['sig'])
+class UnsignedBlock(rlp.Serializable):
+
+    fields = Block.fields[:-1]
```
[the remark this answers is not in the document]
Comment (paulperegud) [reply plasma_core/block.py]: Done. BTW, both ways do not work with pyrlp==1.0.0+.

File: tests/contracts/rlp/test_plasma_core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pgebal) [reply tests/contracts/rlp/test_plasma_core.py]: It's still as it was (that '0x000..'). I accept this PR but please fix that before merging.

**Pull request merged**